homeassistant/components/tedee/config_flow.py

```diff
@@ -45,9 +59,30 @@ async def async_step_user(
             step_id="user",
             data_schema=vol.Schema(
                 {
-                    vol.Required(CONF_HOST): str,
-                    vol.Required(CONF_LOCAL_ACCESS_TOKEN): str,
+                    vol.Required(
+                        CONF_HOST,
+                    ): str,
+                    vol.Required(
+                        CONF_LOCAL_ACCESS_TOKEN,
+                    ): str,
                 }
             ),
             errors=errors,
         )
+
+    async def async_step_reauth(self, entry_data: Mapping[str, Any]) -> FlowResult:
+        """Perform reauth upon an API authentication error."""
+        self.reauth_entry = self.hass.config_entries.async_get_entry(
+            self.context["entry_id"]
+        )
+        return self.async_show_form(
+            step_id="user",
```

Comment:
We shouldn't move directly to the form of another step. The form is private to its step method. We're only allowed to change step by calling that step method.

For reauth flows, normally we implement a confirm step with a dedicated form if the user form can't be used again.

Reply:
ah the classic 3am review, couldn't sleep? 😉 Addressed in #109858